viewer/src/main/java/org/kaazing/monitoring/agrona/viewer/task/MetricsTaskImpl.java

```diff
@@ -13,16 +14,20 @@
  * and adding them directly to the logger output.
  *
  */
-public class MetricsTaskImpl implements MetricsTask{
+public class MetricsTaskImpl implements MetricsTask {
     private String fileName;
     private ScheduledFuture<?> task;
 
-    public MetricsTaskImpl(String fileName, ScheduledExecutorService taskExecutor, MetricsCollector metricsCollector) {
+    public MetricsTaskImpl(String fileName, ScheduledExecutorService taskExecutor, MMFReader reader) {
         this.fileName = fileName;
         task = taskExecutor.scheduleAtFixedRate(() -> {
-            // metrics retrieved - log message directly added to the logger output for each metric (using default logging in metricsCollector.getMetrics() method)
-            metricsCollector.getMetrics();
-        }, 0, MetricsViewer.UPDATE_INTERVAL, TimeUnit.MILLISECONDS);
+            // metrics retrieved - log message directly added to the logger output for each metric (using default
+            // logging in metricsCollector.getServiceCounters() method)
+                for (ServiceCounters service : reader.getServices()) {
+                    reader.getServiceCounters(service);
```

Comment:
Just call service.getCounters() instead of reader.getServiceCounters(service); then you can get rid of the getServiceCounters method.